Use [[deprecated]] instead of C10_DEPRECATED

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30918 Use [[deprecated]] instead of C10_DEPRECATED**

This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

smessmer committed:
Use [[deprecated]] instead of C10_DEPRECATED

This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[deprecated]] instead of C10_DEPRECATED"


This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[deprecated]] instead of C10_DEPRECATED"


This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[deprecated]] instead of C10_DEPRECATED"


This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[deprecated]] instead of C10_DEPRECATED"


This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[deprecated]] instead of C10_DEPRECATED"


This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit d626110:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d626110dda85e6b39999938beab291d70cb133cb).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3960051) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=74831799))
<details>
<summary>
<code>Automatic merge failed; fix conflicts and then commit the result.</code>
</summary>

```
CONFLICT (add/add): Merge conflict in .circleci/validate-docker-version.py 
Auto-merging .circleci/validate-docker-version.py 
CONFLICT (add/add): Merge conflict in .circleci/scripts/should_run_job.sh 
Auto-merging .circleci/scripts/should_run_job.sh 
CONFLICT (add/add): Merge conflict in .circleci/generate_config_yml.py 
Auto-merging .circleci/generate_config_yml.py 
CONFLICT (add/add): Merge conflict in .circleci/docker/common/install_travis_python.sh 
Auto-merging .circleci/docker/common/install_travis_python.sh 
CONFLICT (add/add): Merge conflict in .circleci/config.yml 
Auto-merging .circleci/config.yml 
Automatic merge failed; fix conflicts and then commit the result. 
```

</details>



---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

smessmer committed:
Update on "Use [[deprecated]] instead of C10_DEPRECATED"


This is a C++14 feature we can use now

Differential Revision: [D18869636](https://our.internmc.facebook.com/intern/diff/D18869636/)

[ghstack-poisoned]
